dali/operators/reader/loader/indexed_file_loader.h

```diff
@@ -112,7 +112,7 @@ class IndexedFileLoader : public Loader<CPUBackend, IndexedFileLoaderSample, tru
     }
     next_seek_pos_ = seek_pos + size;
 
-    if (opts.use_mmap && current_file_->CanMemoryMap()) {
+    if (!copy_read_data_) {
       auto p = current_file_->Get(size);
```

Comment:
if the `current_file` is an S3 location, then you will end up trying to call `Get`, which will throw (can't be done for for S3). I believe you should keep the  current_file_->CanMemoryMap() check

Reply:
Ok. If we can mix plain files and S3 objects inside one reader this makes a lot of sense. Removed.